package/MDAnalysis/topology/TOPParser.py

```diff
@@ -620,3 +650,23 @@ def parse_dihedrals(self, diha, dihh):
         dihedrals = Dihedrals(dihed)
         impropers = Impropers(improp)
         return dihedrals, impropers
+
+    def parse_chainids(self, num_per_record, numlines):
+        """Extracts the chainID of each residue
+
+        Parameters
+        ----------
+        num_per_record : int
+            The number of entries for each record in section (unused input)
+        numlines : int
+            The number of lines to be parsed in current section
+
+        Returns
+        -------
+        attr : :class:`Segids`
+            A :class:`Segids` instance containing the chainID of each residue
+            as defined in the parm7 file
+        """
+        vals = self.parsesection_mapper(numlines, lambda x: x)
```

Comment:
Does the file guarantee a CHAINID for each residue or is it possible that some don't have one?

If that could happen, do we provide a default or do we have blank segids?

Reply:
Yes, if not, that's an error, we have no way to know which chainID belongs to which residue if there're skips.

Reply:
Is there a way that we can raise an error if we get an inconsistent file? 

Or at least an `assert` with an associated comment that gets triggered in the unexpected case when a prm file did not correctly include chainids either for everything or for nothing?

Reply:
Regarding this issue, this was my opinion after implementing the fixes:

> Regarding errors in the RESIDUE_CHAINID flag: I decided that giving a proper warning and just reading the file as if there was no RESIDUE_CHAINID info was the right thing to do, mainly because this is an optional flag and I don't think a field that is optional should prevent the user from doing what they want. Let me know if it's ok with you.

https://github.com/MDAnalysis/mdanalysis/pull/4007#issuecomment-1645349308

And [this](https://github.com/MDAnalysis/mdanalysis/blob/53b29aa71c5b6c9776dc4cdee153d9ccdffdaefa/package/MDAnalysis/topology/TOPParser.py#L322) is the warning. Do you want me to turn that into an error?

Reply:
I agree with you with issuing a warning and not failing on an optional part of the file.